File: task_equivalents/python/data_visualization/01_scatter_trend_grouped.py
# Extracted from Ch03.py (Q8 visualization)

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm

Auto = pd.read_csv('data/csv/Auto.csv', na_values='?').dropna()

n = len(Auto['mpg'])
X = np.reshape(Auto['horsepower'], (n, 1))
Y = np.reshape(Auto['mpg'], (n, 1))
X = sm.add_constant(X)
model1_fit = sm.OLS(Y, X).fit()

a = model1_fit.params[0]
b = model1_fit.params[1]
yfit = [a + b * xi for xi in X[:, 1]]

plt.scatter(X[:, 1], Y)
plt.xlabel('horsepower')
plt.ylabel('mpg')
plt.plot(X[:, 1], yfit, 'red')

File: task_equivalents/r/data_visualization/01_scatter_trend_grouped.R
# Extracted from Ch03.R (Q8 visualization)

auto = read.csv('data/csv/Auto.csv', header=TRUE, na.strings='?')
auto = na.omit(auto)

n = nrow(auto)
X = matrix(auto$horsepower, ncol=1)
Y = matrix(auto$mpg, ncol=1)
X = cbind(1, X)
model1 = lm(Y ~ X[,2])

a = coef(model1)[1]
b = coef(model1)[2]
yfit = a + b * X[,2]

plot(X[,2], Y, xlab='horsepower', ylab='mpg')
lines(X[,2], yfit, col='red')
